src/llm/backends/openai.py

```
@@ -240,10 +240,41 @@ async def complete(
+                )
+                try:
+                    content = self._parse_or_repair_structured_content(
+                        json_object_response,
+                        response_format,
+                        model,
+                        empty_on_missing=False,
+                    )
+                except (ValidationException, StructuredOutputError, ValidationError):
+                    # json_object mode also failed; raise the original exception
+                    # so the retry/fallback chain can engage as before.
+                    raise structured_exc
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)